src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1211,6 +1232,461 @@ public static XmlDocument CreateJobXmlDocument(string svrName, string jobName)
+                            ";
+                            List<SqlParameter> notebookDatabaseSetupQueryParams = new List<SqlParameter>();
+                            notebookDatabaseSetupQueryParams.Add(new SqlParameter("jobName", parameters.Notebook.Name));
+                            DataSet jobIdDataSet = ExecuteQuery(connInfo, notebookDatabaseSetupQueryString, notebookDatabaseSetupQueryParams, targetDatabase);
+                            DataTable jobIdDataTable = jobIdDataSet.Tables[0];
+                            DataRow jobIdDataRow = jobIdDataTable.Rows[0];
+                            string jobId = ((Guid)jobIdDataRow["job_id"]).ToString();
+                            templateFileContents = templateFileContents.Replace("'", "''");
+                            string insertTemplateJsonQuery =
+                            @"
+                            INSERT INTO notebooks.nb_template(job_id, notebook, last_run_notebook_error, execute_database) VALUES (@jobId, @templateFileContents, N'', @executeDatabase)
+                            ";
+                            List<SqlParameter> insertTemplateJsonQueryParams = new List<SqlParameter>();
+                            insertTemplateJsonQueryParams.Add(new SqlParameter("jobId", jobId));
+                            insertTemplateJsonQueryParams.Add(new SqlParameter("templateFileContents", templateFileContents));
+                            insertTemplateJsonQueryParams.Add(new SqlParameter("executeDatabase", parameters.Notebook.ExecuteDatabase));
+                            ExecuteQuery(connInfo, insertTemplateJsonQuery, insertTemplateJsonQueryParams, targetDatabase);
+                        }
+                    }
+                    result.Success = true;
+                    result.Job = parameters.Notebook;
+                }
+                catch (Exception e)
+                {
+                    result.Success = false;
+                    result.ErrorMessage = e.ToString();
+                }
+                await requestContext.SendResult(result);
+            });
+        }
+
+        internal async Task HandleDeleteAgentNotebooksRequest(DeleteAgentNotebookParams parameters, RequestContext<ResultStatus> requestContext)
+        {
+            await Task.Run(async () =>
+            {
+                var result = new ResultStatus();
+                try
+                {
+
+                    string deleteNotebookRowQuery =
+                    @"
+                    DELETE FROM notebooks.nb_template
+                    WHERE 
+                    job_id = @jobId;
+                    DELETE FROM notebooks.nb_materialized
+                    WHERE
+                    job_id = @jobId;
+                    IF NOT EXISTS (SELECT * FROM notebooks.nb_template)
+                    BEGIN
+                        DROP TABLE notebooks.nb_template;
+                        DROP TABLE notebooks.nb_materialized;
+                        DROP SCHEMA notebooks;
```

Comment:
this may fail if the user creates new object within here.  is there a way to either (1) check if there's still content here before trying to delete or (2) try\catch this statement or isolate in a different batch such that the drop tables and other cleanup succeeds even when the drop schema fails.